Exercise/compute_lowest_terms.py

```diff
@@ -3,41 +3,54 @@
 def lowest_terms(x):
 	numbers = x.split("/")
 	
+	# Check for zero denominator and zero numerator
 	if numbers[0] == "0": return "0"
 	if numbers[1] == "0": return "Undefined"
 
-	if (int(numbers[0]) < 0 and int(numbers[1]) < 0) or (int(numbers[1]) < 0 and int(numbers[0]) > 0):
+	# converts negative denominator to positive and transfer it to numerator
+	# eliminate negative sign if both the numerator and denominator are negative
+	if (int(numbers[1]) < 0):
 		numbers = [-1 * int(x) for x in numbers]
-	
-	factors_of_numerator = set(factors(numbers[0]))
-	factors_of_denominator = set(factors(numbers[1]))
 
-	common_factor = factors_of_numerator & factors_of_denominator
+	# generate unique set of common factor of the numerator and the denominator
+	common_factor = set(factors(numbers[0])) & set(factors(numbers[1]))
 
+	# check for common factors and get the highest common factor
+	# divide the give fractions with highest common factor and return the results
 	if common_factor:
 		highest_common_factor = max(common_factor)
-
-	if highest_common_factor:
 		return "{}/{}".format(int(numbers[0]) // highest_common_factor, int(numbers[1]) // highest_common_factor )
 
+	# if nothing is found about the numbers, return them as the are given
 	return "{}/{}".format(numbers[0], numbers[1])
 
 
+# ###########################################################################
+# A function that generate the factors of any number
+# It accept a number
+# Returns a list of factors or empty list if there are no factors
+# ###########################################################################
```

Comment:
This should be the first line of the factors() function and properly formatted
check distance_to_destination() function in Examples folder for sample docstring format